File: SmartWash/src/main/java/com/smartwash/common/OrderStatus.java
package com.smartwash.common;

import lombok.Getter;

@Getter
public enum OrderStatus {
    CANCELED("-2", "已取消"),
    REFUNDED("-1", "已退款"),
    PENDING_PAYMENT("0", "待支付"),
    PENDING_SHIPMENT("1", "待寄件"),
    RECEIVED("2", "已收取"),
    WASHING("3", "清洗中"),
    DRIED("4", "已烘干"),
    IN_DELIVERY("5", "配送中"),
    READY_FOR_PICKUP("6", "待取件"),
    COMPLETED("7", "已完成");

    private final String status;
    private final String description;

    OrderStatus(String status, String description) {
        this.status = status;
        this.description = description;
    }
}

File: SmartWash-Android/app/src/main/java/com/smartwash/utils/OrderStatus.kt
package com.smartwash.utils

import androidx.annotation.StringRes
import com.smartwash.R

enum class OrderStatus(val status: String, @StringRes val descriptionRes: Int) {
    CANCELED("-2", R.string.order_status_cancelled),
    REFUNDED("-1", R.string.order_status_refunded),
    PENDING_PAYMENT("0", R.string.order_status_pending_payment),
    PENDING_SHIPMENT("1", R.string.order_status_pending_shipment),
    RECEIVED("2", R.string.order_status_received),
    WASHING("3", R.string.order_status_washing),
    DRIED("4", R.string.order_status_dried),
    IN_DELIVERY("5", R.string.order_status_in_delivery),
    READY_FOR_PICKUP("6", R.string.order_status_ready_for_pickup),
    COMPLETED("7", R.string.order_status_completed);

    companion object {
        private val statusMap = entries.associateBy { it.status }
        fun fromStatus(status: String): OrderStatus? = statusMap[status]
        fun getDescriptionResByStatus(status: String): Int = fromStatus(status)?.descriptionRes ?: R.string.order_status_all
    }
}

enum class ShowOrderStatus(val status: String, @StringRes val descriptionRes: Int) {
    ALL_ORDER("001", R.string.order_status_all),
    PENDING_PAYMENT("0", R.string.order_status_pending_payment),
    PENDING_SHIPMENT("1", R.string.order_status_pending_shipment),
    WASHING("3", R.string.order_status_washing),
    READY_FOR_PICKUP("6", R.string.order_status_ready_for_pickup)
}
